packages/allure-js-commons/src/AllureTest.ts

```diff
@@ -49,4 +50,23 @@ export class AllureTest extends ExecutableItemWrapper {
   addTmsLink(url: string, name: string): void {
     this.addLink(url, name, LinkType.TMS);
   }
+
+  calculateHistoryId(): void {
+    const tcId = this.testResult.testCaseId
```

Comment:
fyi: we have a history ID already exposed: https://playwright.dev/docs/api/class-testcase#test-case-id